include/open62541/plugin/pki.h

```diff
@@ -30,6 +30,33 @@ _UA_BEGIN_DECLS
  * The lifecycle of the plugin is attached to a server or client config. The
  * ``clear`` method is called automatically when the config is destroyed. */
 
+/* Enum and functions for the private key password */
+typedef enum  {
+    UA_PRIVATEKEYPASSWORDSTATE_INITIAL = 0,
+    UA_PRIVATEKEYPASSWORDSTATE_WRONG = 1,
+} UA_PrivateKeyPasswordState;
+
+// Forward declaration for private key encryption
+struct UA_Client;
+
+/* This callback is called when a client with security is initialized with a password
+ * protected private key in the PEM format.
+ * If a wrong password is encountered, the callback will be called repeatedly until
+ * doRetry is set to false. */
+typedef UA_String (*UA_Client_privateKeyPasswordCallback)(UA_Client *client,
```

Comment:
This can be generalized beyond the client.
Also the server can have an encrypted private key.

Reply:
I have restored the generic version with the `void *` context